src/dlsproto/client/internal/SharedResources.d

```diff
@@ -247,5 +280,134 @@ public final class SharedResources
 
             return fiber;
         }
+
+        /**********************************************************************
+
+            Gets a single-shot timer.
+
+            Params:
+                period_s = seconds part of timer interval
+                period_ms = milliseconds part of timer interval
+                timer_dg = delegate to call when timer fires.
+
+            Returs:
+                ITimer interface to a timer to use during the request's lifetime.
+
+        **********************************************************************/
+
+        public ITimer getTimer ( uint period_s, uint period_ms, void delegate ( ) timer_dg )
```

Comment:
Sorry, forgot to mention this before: I'd also suggest just accepting `uint period_ms`. There's no advantage to replicating the annoying internal API of the timer event here, is there?

Reply:
You mean change this to be just:

```
public ITimer getTimer ( uint period_ms, void delegate ( ) timer_dg )
```
?

Reply:
I just preemptively updated it.